src/sql/src/plan/statement/ddl.rs

```diff
@@ -3322,9 +3325,19 @@ pub fn plan_create_cluster(
         }
 
         let disk_default = scx.catalog.system_vars().disk_cluster_replicas_default();
-        let disk = disk.unwrap_or(disk_default);
-        if disk {
-            scx.require_feature_flag(&vars::ENABLE_DISK_CLUSTER_REPLICAS)?;
+        let mut disk = disk_in.unwrap_or(disk_default);
+
+        // HACK(benesch): disk is always enabled for v2 cluster sizes, and it
+        // is an error to specify `DISK = FALSE` or `DISK = TRUE` explicitly.
+        //
+        // The long term plan is to phase out the v1 cluster sizes, at which
+        // point we'll be able to remove the `DISK` option entirely and simply
+        // always enable disk.
+        if is_cluster_size_v2(&size) {
+            if disk_in.is_some() {
+                sql_bail!("DISK option not supported for cluster sizes ending in cc or C");
```

Comment:
This error message will be confusing to users, IMO. _Why_ isn't the option allowed? Maybe we could mention here that cc/C clusters always have disk.

Reply:
Updated with a brief explanation.